### PR TITLE
Start adding max header width

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/panel/issues/4869

Start adding max_header_width, but before continuing, wanted to make sure if this is the desired way of injecting styles inside jinja HTML templates instead of css.

I also removed the shadows (or glow)
image

Because it looked weird.
image

I guess I could keep the shadow if I created another div?

### CODE REVIEW

File: panel/template/fast/list/fast_list_template.html [hunk 1]
```diff
@@ -84,7 +84,7 @@
 <fast-design-system-provider id="body-design-provider" use-defaults>
   <div id="container">
     <fast-design-system-provider id="header-design-provider">
-      <nav id="header" {{ 'class="shadow"' if style.shadow else '' }} >
+      <nav id="header" {% if header_max_width %}style="margin-left: auto; margin-right: auto; max-width: {{ header_max_width }}"{% endif %} >
```
Comment (philippjfr) [inline panel/template/fast/list/fast_list_template.html:87]: Would indeed prefer a CSS class.

Comment (ahuang11) [reply panel/template/fast/list/fast_list_template.html:87]: Should I also migrate the `main_max_width` or perhaps just header_max_width?

**Comment by ahuang11** [comment]
Okay, I couldn't add it to an actual .css file, but I was able to add it to styles.

**Comment by ahuang11** [comment]
image

image

image

image

image

image

image

**Comment by ahuang11** [comment]
Very stale at this point.

**Pull request closed without merging**